lib/python/qmk/cli/flash.py

```diff
@@ -0,0 +1,79 @@
+"""Compile and flash QMK Firmware
+
+You can compile a keymap already in the repo or using a QMK Configurator export.
+A bootloader must be specified.
+"""
+import os
+import sys
+import subprocess
+from argparse import FileType
+
+from milc import cli
+from qmk.commands import create_make_command
+from qmk.commands import parse_configurator_json
+from qmk.commands import compile_configurator_json
+
+import qmk.path
+
+
+def print_bootloader_help():
+    """Prints the available bootloaders listed in docs.qmk.fm.
+
+    Called when 'qmk flash' is called with no keyboard, keymap or file, and a string satisfies the required bootloader argument. This should really be listed 'qmk flash -h'
+    """
+    cli.log.error('You must supply a configuration file and bootloader,')
+    cli.echo('or a -kb keyboard, -km keymap, and bootloader.')
+    cli.echo('For example: qmk flash -kb planck/rev6 -km default dfu-util')
+    cli.echo('or qmk flash 1upkeyboards_60hse_default.json dfu-util')
+    cli.echo('For more info, visit https://docs.qmk.fm/#/flashing')
+    cli.echo('Bootloaders:')
+    cli.echo('dfu')
+    cli.echo('dfu-ee')
+    cli.echo('dfu-split-left')
+    cli.echo('dfu-split-right')
+    cli.echo('avrdude')
+    cli.echo('BootloadHID')
+    cli.echo('dfu-util')
+    cli.echo('dfu-util-split-left')
+    cli.echo('dfu-util-split-right')
+    cli.echo('st-link-cli')
+
+@cli.argument('bootloader', arg_only=True, help='The flash command, corresponding to qmk\'s make options of bootloaders.')
+@cli.argument('filename', nargs='?', arg_only=True, help='The configurator export JSON to compile. Use this if you dont want to specify a keymap and keyboard.')
+@cli.argument('-km', '--keymap', help='The keymap to build a firmware for. Use this if you dont have a configurator file. Ignored when a configurator file is supplied.')
+@cli.argument('-kb', '--keyboard', help='The keyboard to build a firmware for. Use this if you dont have a configurator file. Ignored when a configurator file is supplied.')
```

Comment:
If you want to add a `-h` or a `-t --targets`, you can do the follow
```suggestion
@cli.argument('-b, '--bootloaders', action='store_true', help='List the available bootloaders.')
```

This will provide you with `cli.args.bootloaders` that will be true or false, which then you can lead off with to do something like

```suggestion
if cli.args.bootloaders:
    print_bootloader_help()
    return False
```